heroku/resource_heroku_pipeline_config_var.go

```diff
@@ -0,0 +1,224 @@
+	dupeErr := duplicateVarsChecker(vars, sensitiveVars)
+	if dupeErr != nil {
+		return dupeErr
+	}
+
+	// Combine both sensitive and non-sensitive vars
+	combinedVars := mergeVars(vars, sensitiveVars)
+
+	log.Printf("[INFO] Creating pipeline [%s] stage [%s] config vars", pipelineID, pipelineStage)
+
+	// Update the vars for the pipeline
+	updateErr := updatePipelineConfigVars(client, pipelineID, pipelineStage, nil, combinedVars)
+	if updateErr != nil {
+		return updateErr
+	}
+
+	log.Printf("[INFO] Created pipeline [%s] stage [%s] config vars", pipelineID, pipelineStage)
+
+	// Set the ID to be pipeline ID + stage
+	d.SetId(fmt.Sprintf("%s:%s", pipelineID, pipelineStage))
+
+	return resourceHerokuPipelineConfigVarRead(d, meta)
+}
+
+func resourceHerokuPipelineConfigVarUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).Api
+	pipelineID := getPipelineID(d)
+	pipelineStage := getPipelineStage(d)
+
+	var oldVars, newVars, oldSensitiveVars, newSensitiveVars, allOldVars, allNewVars map[string]interface{}
+	oldVars, newVars = getVarDiff(d, "vars")
+	oldSensitiveVars, newSensitiveVars = getVarDiff(d, "sensitive_vars")
+
+	// Merge the vars
+	allOldVars = mergeVars(oldVars, oldSensitiveVars)
+	allNewVars = mergeVars(newVars, newSensitiveVars)
+
+	log.Printf("[INFO] Updating pipeline [%s] stage [%s] config vars", pipelineID, pipelineStage)
+
+	// Update the vars for the pipeline
+	updateErr := updatePipelineConfigVars(client, pipelineID, pipelineStage, allOldVars, allNewVars)
+	if updateErr != nil {
+		return updateErr
+	}
+
+	log.Printf("[INFO] Updated pipeline [%s] stage [%s] config vars", pipelineID, pipelineStage)
+
+	return resourceHerokuPipelineConfigVarRead(d, meta)
+}
+
+func resourceHerokuPipelineConfigVarRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).Api
+
+	// Parse the resource ID to return the pipeline ID & stage
+	pipelineID, pipelineStage, parseErr := parseCompositeID(d.Id())
+	if parseErr != nil {
+		return parseErr
+	}
+
+	remotePipelineVars, getErr := client.PipelineConfigVarInfoForApp(context.TODO(), pipelineID, pipelineStage)
+	if getErr != nil {
+		return getErr
+	}
+
+	// Need to convert remotePipelineVars to a data type required by vetVarsForState
+	rpvFormatted := make(map[string]string)
+	for key, value := range remotePipelineVars {
+		rpvFormatted[key] = *value
+	}
+
+	vettedConfigVars, vettedSensitiveConfigVars := vetVarsForState(getVars(d), getSensitiveVars(d), rpvFormatted)
+
+	log.Printf("[DEBUG] pipeline config vars to be set in state: *%#v", vettedConfigVars)
+	log.Printf("[DEBUG] pipeline sensitive config vars to be set in state: *%#v", vettedSensitiveConfigVars)
+
+	var setErr error
+	setErr = d.Set("pipeline_id", pipelineID)
+	setErr = d.Set("pipeline_stage", pipelineStage)
+	setErr = d.Set("vars", vettedConfigVars)
+	setErr = d.Set("sensitive_vars", vettedSensitiveConfigVars)
+	setErr = d.Set("all_vars", rpvFormatted)
```

Comment:
Maybe I misunderstand something about Go here, but I think this way of re-assigning the error in succession would hide a lot of error conditions, as long as the last one succeeds. Each one needs to return if error.

```suggestion
	setErr = d.Set("pipeline_id", pipelineID)
	if setErr != nil {
		return setErr
	}
	setErr = d.Set("pipeline_stage", pipelineStage)
	if setErr != nil {
		return setErr
	}
	setErr = d.Set("vars", vettedConfigVars)
	if setErr != nil {
		return setErr
	}
	setErr = d.Set("sensitive_vars", vettedSensitiveConfigVars)
	if setErr != nil {
		return setErr
	}
	setErr = d.Set("all_vars", rpvFormatted)
	if setErr != nil {
		return setErr
	}
```

Reply:
Yeah..you're right. I was a bit lazy and I think `d.Set` has a built-in panic already:
```
	err := d.setWriter.WriteField(strings.Split(key, "."), value)
	if err != nil && d.panicOnError {
		panic(err)
	}
```

so I didn't feel the need to actually define `if setErr != nil { return setErr }` for each `d.Set`. Thoughts?

Reply:
I prefer the style of simply not assigning, to avoid the visual noise:

```go
d.Set("pipeline_id", pipelineID)
```

But with built-in panics and the convention to not check `Set` errors, this is fine.